src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * @param collectionName
+ * @param vectorEnabled
+ * @param vectorSize
+ * @param similarityFunction
+ * @param vectorizeServiceName
+ * @param modelName
+ */
+public record CollectionSettings(
+    String collectionName,
+    Boolean vectorEnabled,
+    int vectorSize,
+    SimilarityFunction similarityFunction,
+    String vectorizeServiceName,
+    String modelName) {
+
+  /**
+   * The similarity function used for the vector index. This is only applicable if the vector index
+   * is enabled.
+   */
+  public enum SimilarityFunction {
+    COSINE,
+    EUCLIDEAN,
+    DOT_PRODUCT,
+    UNDEFINED;
+
+    public static SimilarityFunction fromString(String similarityFunction) {
+      if (similarityFunction == null) return UNDEFINED;
+      return switch (similarityFunction.toLowerCase()) {
+        case "cosine" -> COSINE;
+        case "euclidean" -> EUCLIDEAN;
+        case "dot_product" -> DOT_PRODUCT;
+        default -> throw new JsonApiException(
+            ErrorCode.VECTOR_SEARCH_INVALID_FUNCTION_NAME,
+            ErrorCode.VECTOR_SEARCH_INVALID_FUNCTION_NAME.getMessage() + similarityFunction);
+      };
+    }
+  }
+
+  public static CollectionSettings getVectorProperties(
+      Schema.CqlTable table, ObjectMapper objectMapper) {
+    String collectionName = table.getName();
+    final Optional<QueryOuterClass.ColumnSpec> first =
+        table.getColumnsList().stream()
+            .filter(
+                c -> c.getName().equals(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME))
+            .findFirst();
+    boolean vectorEnabled = first.isPresent();
+    if (vectorEnabled) {
+      final int vectorSize = first.get().getType().getVector().getSize();
+      final Optional<Schema.CqlIndex> vectorIndex =
+          table.getIndexesList().stream()
+              .filter(
+                  i ->
+                      i.getColumnName()
+                          .equals(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME))
+              .findFirst();
+      CollectionSettings.SimilarityFunction function = CollectionSettings.SimilarityFunction.COSINE;
+      if (vectorIndex.isPresent()) {
+
+        if (vectorIndex
+            .get()
+            .getOptionsMap()
+            .containsKey(DocumentConstants.Fields.VECTOR_INDEX_FUNCTION_NAME)) {
+          function =
+              CollectionSettings.SimilarityFunction.fromString(
+                  vectorIndex
+                      .get()
+                      .getOptions()
+                      .get(DocumentConstants.Fields.VECTOR_INDEX_FUNCTION_NAME));
+        }
+      }
+      final String comment = table.getOptionsOrDefault("comment", null);
+      if (comment != null && !comment.isBlank()) {
+        try {
+          JsonNode vectorizeConfig = objectMapper.readTree(comment);
+          String vectorizeServiceName =
+              vectorizeConfig != null && vectorizeConfig.has("service")
```

Comment:
I think `readTree()` should never return `null`; if content missing, should throw some parsing exception.
Because of this, none of null checks for `vectorizeConfig` should be needed.

If `null` could be returned, could use:

    if (vectorizeConfig == null) {
       vectorizeConfig = objectMapper.missingNode();
    }